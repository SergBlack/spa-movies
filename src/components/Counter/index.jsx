import React from 'react';
import { number, string } from 'prop-types';
import styled from 'styled-components';

const StyledCounter = styled.div`
  margin: 16px 48px;
  font-size: 24px;
`;

const Counter = ({ count, text }) => (
  <StyledCounter>
    <strong>{count}</strong>
    {' '}
    <span>{text}</span>
  </StyledCounter>
);

Counter.propTypes = {
  count: number.isRequired,
  text: string.isRequired,
};

export default Counter;
