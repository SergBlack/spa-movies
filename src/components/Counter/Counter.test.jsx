import React from 'react';
import renderer from 'react-test-renderer';

import Counter from './index';

it('render with props', () => {
  const tree = renderer.create(
    <Counter
      count={42}
      text="Answer to the Ultimate Question of Life, the Universe, and Everything"
    />,
  );
  expect(tree).toMatchSnapshot();
});
