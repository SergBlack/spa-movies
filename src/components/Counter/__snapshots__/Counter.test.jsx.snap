// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with props 1`] = `
<div
  className="sc-bdfBwQ kSGToi"
>
  <strong>
    42
  </strong>
   
  <span>
    Answer to the Ultimate Question of Life, the Universe, and Everything
  </span>
</div>
`;
